front/lib/api/assistant/generation.ts

```diff
@@ -406,10 +407,11 @@ export async function renderConversationForModelMultiActions({
       if (contentRes.isErr()) {
         return new Err(contentRes.error);
       }
-      selected.unshift({
-        ...msg,
-        content: contentRes.value + truncationMessage,
-      });
+
+      const previousMessage = selected[0];
+      previousMessage.content =
+        // The truncation message is outside of the xml tag <file-attachement/>
```

Comment:
so you are not counting the actual XML tags as tokens ?

Reply:
They are added in the previous loop so they are accounted for yes 👍